### PR TITLE
Added Binary Search [Go]

### LINKED ISSUE #237: Binary Search [Go]
https://en.wikipedia.org/wiki/Binary_search_algorithm
Assign me

### PR DESCRIPTION
**Pull request by aashutoshrathi**
Fixes #237 

#### By submitting this pull request I confirm I've read and complied with the below declarations.

- [x] I have read the [Contribution guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [coding guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#cs) for this project.
- [x] My code follows the [skeleton code structure](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#sample).
- [x] This pull request has a descriptive title. For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [x] This pull request will be closed if I fail to update it even once in a continuous time span of 7 days.

### CODE REVIEW

File: binary_search/binarySearch.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
+package main
+
+import "fmt"
+
+func binary_search(array []int, target int) int {
+	left := 0
+	right := len(array) - 1
```
Comment (mohitkyadav) [inline binary_search/binarySearch.go]: `left` and `right` both can be declared in a single line.

File: binary_search/binarySearch.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
+package main
+
+import "fmt"
+
+func binary_search(array []int, target int) int {
```
Comment (singhpratyush) [inline binary_search/binarySearch.go]: Actually, it should be `BinarySearch`.

File: binary_search/binarySearch.go [hunk 3]
```diff
@@ -0,0 +1,30 @@
+package main
+
+import "fmt"
+
```
Comment (mohitkyadav) [inline binary_search/binarySearch.go]: Add function comments

File: binary_search/binarySearch.go (hunk 2 above)
Comment (aashutoshrathi) [inline binary_search/binarySearch.go]: @singhpratyush it should `binarySearch`, isn't it?

File: binary_search/binarySearch.go [hunk 4]
```diff
@@ -0,0 +1,31 @@
+package main
+
```
Comment (aviaryan) [inline binary_search/binarySearch.go]: Filename should be all smalls. Make it binary_search.go

File: binary_search/binarySearch.go [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+	return -1
+}
+
+func main() {
+	array := []int{1, 5, 35, 112, 258, 324, 456, 512}
+	index := BinarySearch(array, 112)
+	if index == -1 {
+		fmt.Println("Element is not present in array")
+	} else {
+		fmt.Println("Element is present at index : ", index)
```
Comment (aviaryan) [inline binary_search/binarySearch.go]: Remove space after `index :`

**Pull request merged**